tests/test_raft.rs

```diff
@@ -248,7 +250,7 @@ impl Network {
     // A nil node will be replaced with a new *stateMachine.
     // A *stateMachine will get its k, id.
     // When using stateMachine, the address list is always [1, n].
-    pub fn new(mut peers: Vec<Option<Interface>>) -> Network {
+    pub fn new(mut peers: Vec<Option<Interface>>, prevote: bool) -> Network {
```

Comment:
Is there any way to map the `new...WithConfig` for `Network`? IMO it's better to able to customize the whole config rather than just `pre_vote`. And it's better to write the code as same as etcd/raft.